sdk/core/azure-core/src/environment.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/core/platform.hpp"
+
+#include "azure/core/internal/environment.hpp"
+
+#if defined(AZ_PLATFORM_WINDOWS)
+#if !defined(WIN32_LEAN_AND_MEAN)
+#define WIN32_LEAN_AND_MEAN
+#endif
+#if !defined(NOMINMAX)
+#define NOMINMAX
+#endif
+
+#include <vector>
+
+#include <windows.h>
+#else
+#include <stdlib.h>
+#endif
+
+using Azure::Core::_internal::Environment;
+
+std::string Environment::GetVariable(const char* name)
+{
+  if (name != nullptr && name[0] != 0)
+  {
+#if defined(AZ_PLATFORM_WINDOWS)
+    std::vector<std::string::value_type> bufferVector;
+    std::string::value_type* buffer = nullptr;
+    DWORD bufferSize = 0;
+    while (const auto requiredSize = GetEnvironmentVariableA(name, buffer, bufferSize))
+    {
+      if (requiredSize < bufferSize)
+      {
+        return std::string(buffer, buffer + (bufferSize - 1));
+      }
+
+      bufferVector.resize(static_cast<decltype(bufferVector)::size_type>(requiredSize));
+      bufferSize = requiredSize;
+      buffer = bufferVector.data();
+    }
+#else
+    if (const auto value = getenv(name))
+    {
+      return std::string(value);
+    }
+#endif
+  }
+  return std::string();
+}
+
+void Environment::SetVariable(const char* name, const char* value)
```

Comment:
the SDK should never set env vars.  We should not have this function or any code setting env vars within the samples or anything.
env is expected to be set by the user / customer.

Reply:
Tests set them.

Reply:
neither tests or SDK should.  That's unexpected if I set my env and some test changed it

Reply:
Tests need to be able to set env variables, to test the code that relies on env variables: console logger, environment credential, managed identity credential. But they also store previous value, and revert it back to original. And even if something went wrong, the variables you set/modify only apply to current process and its children, i.e. basically only the unit test exe while it is running.

Reply:
E.G. Logger tests in azure core need to set the ENVs in order to  test that the logger properly respects ENV values when logging messages with different levels